runtime/sema/type.go

```diff
@@ -5192,6 +5302,10 @@ func (t *ReferenceType) RewriteWithRestrictedTypes() (Type, bool) {
 	}
 }
 
+func (t *ReferenceType) Map(gauge common.MemoryGauge, typeParamMap map[*TypeParameter]*TypeParameter, f func(Type) Type) Type {
+	return f(NewReferenceType(gauge, t.Type.Map(gauge, typeParamMap, f), t.Authorization))
```

Comment:
There's a lot of nesting going on here, maybe break this down a bit to make it easier to read:
```suggestion
	mappedType := t.Type.Map(gauge, typeParamMap, f)
	return f(NewReferenceType(
		gauge, 
		mappedType, 
		t.Authorization,
	))
```